Release notes for 2.14.19 (#11804)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.14.19

## Dynamic descriptions for paused and suspended flow runs

> You can now include dynamic, markdown-formatted descriptions when pausing or suspending a flow run for human input. This description will be shown in the Prefect UI alongside the form when a user is resuming the flow run. Enabling developers to give context and instructions to users when they need to provide input.

Reviewer: ```suggestion
You can now include dynamic, markdown-formatted descriptions when pausing or suspending a flow run for human input. This description will be shown in the Prefect UI alongside the form when a user is resuming the flow run, enabling developers to give context and instructions to users when they need to provide input.
```

```python
from datetime import datetime
from prefect import flow, pause_flow_run, get_run_logger
from prefect.input import RunInput

class UserInput(RunInput):
    name: str
    age: int

@flow
async def greet_user():
    logger = get_run_logger()
    current_date = datetime.now().strftime("%B %d, %Y")

    description_md = f"""
**Welcome to the User Greeting Flow!**
Today's Date: {current_date}

Please enter your details below:
- **Name**: What should we call you?
- **Age**: Just a number, nothing more.
"""

    user_input = await pause_flow_run(
        wait_for_input=UserInput.with_initial_data(
            description=description_md, name="anonymous"
        )
    )

    if user_input.name == "anonymous":
        logger.info("Hello, stranger!")
    else:
        logger.info(f"Hello, {user_input.name}!")
```

> 

Reviewer 2: I'm not sure we have a hard rule for this but I've liked the touch of referencing the implementation PRs for the 
highlights. For this highlight I think these are the ones we'd want to call out:

- (implementation) - https://github.com/PrefectHQ/prefect/pull/11776
-  (docs) https://github.com/PrefectHQ/prefect/pull/11799

Author: Good call!

### Enhancements

> - Enhanced RunInput saving to include descriptions, improving clarity and documentation for flow inputs — https://github.com/PrefectHQ/prefect/pull/11776

Reviewer 2: ```suggestion
- Enhanced `RunInput` saving to include descriptions, improving clarity and documentation for flow inputs — https://github.com/PrefectHQ/prefect/pull/11776
```

- Improved type hinting for automatic run inputs, enhancing the developer experience and code readability — https://github.com/PrefectHQ/prefect/pull/11796
- Extended Azure filesystem support with the addition of `azure_storage_container` for more flexible storage options — https://github.com/PrefectHQ/prefect/pull/11784
- Added deployment details to work pool information, offering a more comprehensive view of work pool usage — https://github.com/PrefectHQ/prefect/pull/11766

### Fixes

> - Updated UI to make links within panels interactive, enhancing user navigation and experience — https://github.com/PrefectHQ/prefect/pull/11774

Reviewer 2: ```suggestion
- Updated terminal based deployment operations to make links within panels interactive, enhancing user navigation and experience — https://github.com/PrefectHQ/prefect/pull/11774
```

Reviewer 2: I think it would be worth making a distinction that the interface here refers to a terminal.
